### ISSUE TITLE
UpheldBy restarts unit even when manually stopped

### ISSUE THREAD

**Issue by aksiksi** [systemd/systemd#35636]
### Component

systemctl, systemd

### Is your feature request related to a problem? Please describe

Suppose I have two units `db.service` and `app.service`, where `app` has:

```
After=db.service
Requires=db.service
UpheldBy=db.service
```

The reason for this config is to ensure that `app` is stopped when `db` is stopped, but also restarted when `db` is restarted.

However, it seems `UpheldBy` also applies if `app` is **manually stopped** via `systemctl`.

This behavior is surprising to me, but maybe intentional. Or perhaps systemd doesn't differentiate between manual and automated in general.

### Describe the solution you'd like

A more concrete example where I have two Podman containers running as systemd services:

```
$ sudo systemctl list-dependencies podman-tubesync.service
podman-tubesync.service
...
● ├─podman-tubesync-db.service
```

If I manually stop `podman-tubesync.service`, I'd expect it to stay stopped until started again (manually) rather than continually restart due to `UpheldBy=podman-tubesync-db.service`.

### The systemd version you checked that didn't have the feature you are asking for

256.8

**Comment by rnhmjoj** [comment]
Why was this issue closed? It looks like it is still a problem: I can reproduce in 256.8 and I don't see a fix mentioned anywhere.

**Comment by aksiksi** [comment]
I wasn’t sure whether this was a bug or expected behavior, and instead sent an email to the mailing list asking about this (which got no reply).

**Comment by YHNdnzj** [comment]
This is very much the semantics of Upholds=... As long as any unit upholding it is active it would always be restarted, no other factors affect this.

Not a bug, closing.

**Comment by aksiksi** [comment]
Thank you for clarifying - I assumed that was the case. 

Upholds is an extremely useful feature, but this behavior confused me quite a bit. I also think having the ability to “override” Upholds and actually (manually) stop a unit is useful.

Would it be possible to differentiate internally between a manual unit stop and a stop due to unit failure (e.g., retries exhausted)? To me, the latter is where it makes sense for the parent to “revive” the child unit.

**Comment by kj** [comment]
I agree. Even `Restart=always` makes the distinction for a user's intentional command. From `systemd.service(5)`:

> When the death of the process is a result of systemd operation (e.g. service stop or restart), the service will not be restarted.

It seemed more intuitive to me that `UpheldBy=` would behave the same way.
